src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,473 @@
+when using `async` and `await`: 
+* __The `await` keyword works only in `async` functions.__
+* __To define an `async` function, add `async` before the function body.__
+
+Here's an example  that converts `main()` from a synchronous to asynchronous function.
+
+First, add the `async` keyword before the function body.
+{% prettify dart %}
+    main() [!async!] {
+{% endprettify %}
+
+If needed, update the function's type signature to return a future.
+{% prettify dart %}
+    [!Future<void>!] main() async {
+{% endprettify %}
+
+Remember, you can only use the `await` keyword within an `async` function. 
+The following examples provide a comparison of synchronous and asynchronous functions.
+If your window is wide enough, you'll see the two examples side-by-side.
+
+<div class="container">
+  <div class="row">
+    <div class="col-sm">
+    <h4> Example: synchronous functions</h4>
+
+{% prettify dart %}
+// Synchronous
+String createOrderMessage () {
+  var order = getUserOrder();
+  return 'Your order is: $order';
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is
+  // more complex and slow
+  return 
+    Future.delayed(
+      Duration(seconds: 4), () => 'Large Latte'); 
+}
+
+// Synchronous
+main() {
+  print('Fetching user order...');
+  print(createOrderMessage());
+}
+
+// "Your order is: Instance of '_Future<String>'"
+{% endprettify %}
+    </div>
+    <div class="col-sm">
+    <h4> Example: asynchronous functions</h4>
+
+{% prettify dart %}
+// Asynchronous
+[!Future<String>!] createOrderMessage () [!async!] {
+  var order = [!await!] getUserOrder();
+  return 'Your order is: $order';
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is
+  // more complex and slow
+  return
+   Future.delayed(
+     Duration(seconds: 4), () => 'Large Latte');
+}
+
+// Asynchronous
+main() [!async!] {
+  print('Fetching user order...');
+  print([!await!] createOrderMessage());
+}
+
+// "Your order is: Large Latte"
+{% endprettify %}
+    </div>
+  </div>
+</div>
+
+The asynchronous example is different in three ways:
+  * The return type for `createOrderMessage()` changes from `String` to `Future<String>`.
+  * The **`async`** keyword appears before the function bodies for `createOrderMessage()` and the `main()` function.
+  * The **`await`** keyword appears before calling the asynchronous functions `getUserOrder()` and `createOrderMessage()`.
+
+{{ site.alert.secondary }}
+  **Key terms:**
+* **async**: You can use the `async` keyword before a function's body to mark it as
+asynchronous.
+* **async function**:  An `async` function is a function labeled with the `async` keyword. 
+* **await**: You can use the `await` keyword to get the completed result of an
+asynchronous expression. You can only use the `await` keyword within an `async` function. 
+{{ site.alert.end }}
+
+## Execution flow with async and await
+
+As of Dart 2, functions marked as `async` run synchronously until the first
```

Comment:
Sure, done. 

IMHO the styling for these notes is too loud -- it adds a lot of a visual noise to the page, both because of the margins (eats up screen real estate *really* quickly and because of distracting bg color it adds. I try to avoid using our site.alert  at all costs.

Reply:
If it looks bad, let's not do it. We've just had experience that having "As of Dart 2" wording confuses readers, so I'd like to avoid doing that. But I'd also like to avoid implying that it's always been this way, so that people who knew the first behavior don't get confused.